src/Features/LanguageServer/Protocol/Handler/RequestExecutionQueue.RequestTelemetryLogger.cs

```diff
@@ -52,28 +62,57 @@ public void LogFailure(string methodName)
             /// </summary>
             public void Dispose()
             {
+                if (_histogramLogAggregator is null || _histogramLogAggregator.IsEmpty)
+                {
+                    return;
+                }
+
+                var queuedDurationCounter = _histogramLogAggregator.GetValue(QueuedDurationKey);
+                Logger.Log(FunctionId.LSP_TimeInQueue, KeyValueLogMessage.Create(LogType.Trace, m =>
+                {
+                    m["server"] = _serverTypeName;
+                    m["bucketsize"] = queuedDurationCounter?.BucketSize;
+                    m["maxbucketvalue"] = queuedDurationCounter?.MaxBucketValue;
+                    m["buckets"] = queuedDurationCounter?.GetBucketsAsString();
+                }));
+
                 foreach (var kvp in _requestCounters)
                 {
+                    var requestExecutionDuration = _histogramLogAggregator.GetValue(kvp.Key);
+
                     Logger.Log(FunctionId.LSP_RequestCounter, KeyValueLogMessage.Create(LogType.Trace, m =>
```

Comment:
Yeah - so this functionid was already gdpr approved, so rather than create a new one I thought it would be easier to re-use.  Perhaps I should rename it though to something like RequestMetrics?

I can also update the property names to be more descriptive

Reply:
please don't reuse.  let's start fresh with a set of IDs and whatnot that we know are good, are crisply defined, and are associated with an owner who has to track this stuff and set appropriate alarms and whatnot.